### DISCUSSION TITLE
Looking for excuse to use Stride3D

### DISCUSSION THREAD

**Discussion by najak3d** [stride3d/stride#1289] [General]
We are currently using Urho3D.NET for our app.   Their lack of Impostor support has sent me looking for alternate solutions.

We are also starting an effort of trying to accomplish the following for MAUI:
1. Provide an easy-to-use lightweight 2D/3D rendering solution that runs inside of a Maui UI (on a picture-box-like control; ideally an optimized Stride3D render surface).
2. Provide a way for those who prefer the MAUI UI to render this UI inside of a Game window.   (We're doing this now with Urho.NET with the Avalonia UI.)

Our default choice, currently, was Urho.NET, because so much progress has been achieved already.   But the cons of Urho are:

1. C++ engine wrapped in C# -- harder to debug when something goes wrong.
2. Non-Mainstream engine, although it's community seems stronger than that of Stride3D.

For Stride3D, we can overcome #.2 above, by finding ways to bring in a larger audience, which can be achieved in-part by integrating with MAUI and Avalonia (both ways).   This'll make it a natural choice for Xamarin/Maui devs to make easy use of Stride3D.

We've narrowed our choice down to:
1. Urho.NET - because we're using it already, and have made considerable progress.
2. Stride3D - because it appears to be the only pure-C# viable candidate.
3. Unity3D - because it by far has the largest community and plethora of add-ons, etc.   BUT, it's commercialized, and not open-source, so if we have issue with integrating to/from MAUI, we have no ability to augment source code, etc.   And ALSO it's closed-source C++, which means that you can't debug when things go wrong.

Stride3D is set up to be the winner here, IMO, because of it being open-source pure C#.   But I think it has the least momentum and the smallest community, currently.   So this is a big concern, and might disqualify Stride3D as a viable contender.

I'm posting here, so that the community/contributors can correct me where I'm wrong, or let us know if they believe they can accomplish this MAUI integration, which for our purposes, is a tantamount.   I am willing to do some heavy-lifting here too, if I become convinced that Stride3D is worth my investment of time.

**Comment by xen2** [comment]
Thanks for your feedback and interest!
Keep us posted on your decision, and don't hesitate if you think of a way that could bring some synergy between your needs/team and the OSS development (i.e. some of the MAUI stuff you explained in https://github.com/stride3d/stride/discussions/1031#discussioncomment-2024945), we could discuss it further!

**Reply by xen2** [reply]
To clarify, I didn't say MAUI will be the way to go for sure (it could be, but there are pros/cons and other options too).
This choice will probably be done in #1031

**Reply by najak3d** [reply]
MAUI is currently a TBD.   

Regarding the concept of enabling MAUI to render inside the Stride3D game window, much depends upon if they can deliver, and also answer our questions regarding the support for plugging in a custom Rendering platform - such that we can render the MAUI output directly to a graphics Texture on the GPU, which then will be rendered like a 2D sprite in Stride.  (This works perfectly in Avalonia+Urho now.)

We've decide to put some faith into Stride3D now, primarily because of the recent "Code-Only" samples and focus of this engine.  It looks like they are opening the doors wide for Code-Only solutions, making Stride3D highly usable/flexible for many more application types; and this is gives us cause for great optimism.  I think our switch over to Stride3D looks far more likely now, than I originally thought when I made this post.

**Comment by HydeNaut** [comment]
Regarding your Discord posts about stride there is this project that I believe is doing much of what you're interested in:
https://github.com/TheKeyblader/Stridelonia

**Reply by najak3d** [reply]
Yes, looks like he's crossed some big bridges here!   That may be all we need to give us our excuse to swap engines.

My short-term plan is to finish up my Urholonia POC, nearing completion.   Then might switch over and try to accomplish the same thing using Stridelonia.
